src/System.Console/tests/WindowAndCursorProps.cs

```diff
@@ -31,6 +31,15 @@ public static void BufferSize_GettingSameAsWindowSize()
     }
 
     [Fact]
+    [SkipOnTargetFramework(TargetFrameworkMonikers.Netcoreapp | TargetFrameworkMonikers.NetcoreUwp)]
+    public static void WindowWidth_WindowHeight_InvalidSize_net46()
+    {
+        Assert.Throws<IOException>(() => Console.WindowWidth = 0);
+        Assert.Throws<IOException>(() => Console.WindowHeight = 0);
+    }
+
+    [Fact]
+    [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework)]
```

Comment:
There are a few ways we've changed exceptions:
1. Improving exception messages.
2. Changing parameter names in ArgumentExceptions.
3. If desktop throws exception A for a condition, throwing exception B where B derives from A for that same condition.
4. If desktop throws exception A for a condition, throwing exception B where B is unrelated to A for that same condition.

I think we can all agree that case (1) makes complete sense and should not be reverted.

Similarly, while case (2) could potentially break someone, the value it provides towards developer productivity is I believe worth it, and the chances that it actually breaks code is low: code would need to be catching ArgumentExceptions (which are usage errors that shouldn't in general be thrown/caught in production code anyway), examining the ParamName, and making a decision based on it.  So I think these should not be reverted, either.

And case (3) has been documented to not be an official breaking change; while it's possible it could break code doing explicit type comparisons on exceptions, we've always reserved the right to return/throw more derived types in subsequent versions.  So we shouldn't revert these, either.

Case (4) is the hard one, and I think we need to demonstrate significant value to change exception types like this, as it can easily break code, e.g. catch blocks for a specific exception type will no longer execute for that condition.  Places where we've reverted exception-related changes thus far have all been this case, and I expect there are more that should be reverted.  Of course, if we've already shipped the revised exception type in .NET Core 1.x, then we have to decide between .NET Core 2.0 maintaining compatibility with .NET Framework vs with .NET Core 1.x, but given that these types have been around in the .NET Framework for over a decade, I'd err on the side of using the same exceptions from .NET Framework, unless there's a really compelling and important reason to do otherwise.